### PR TITLE
Add rating scale question block

### PR DESCRIPTION
**Pull request by CGastrell**
This PR introduces the new Rating Scale question block.

The block still lacks a myriad features and style adjustments, it works in conjunction with D76521-code to handle question creation and responses on the backend.

For the scope of the new block this PR should suffice, further customizations and features to be added on upcoming PRs

## Test instructions
Apply backend patch and checkout this branch. Create/edit a project and add a Rating block

### CODE REVIEW

File: apps/project-renderer/src/components/app/index.js [hunk 1]
```diff
@@ -32,9 +34,7 @@ const App = ( {
 	const [ content, setContent ] = useState( [] );
 	// eslint-disable-next-line
```
Comment (ice9js) [inline apps/project-renderer/src/components/app/index.js:35]: Let's drop this one as well while we're at it?

File: packages/block-editor/src/rating-scale-answer/edit-button.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	onChange,
+	onReplace,
+	onSplit,
+	onDelete,
+} ) => {
+	return (
+		<Button attributes={ attributes } as={ 'div' } className={ className }>
+			<ButtonContent>
+				<RichText
+					placeholder={ __( 'Enter an answer', 'blocks-editor' ) }
```
Comment (ice9js) [inline packages/block-editor/src/rating-scale-answer/edit-button.js:36]: Typo: the i18n domain is `block-editor` not `blocks-editor`.

File: packages/block-editor/src/rating-scale-answer/edit-checkbox.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	onReplace,
+	onSplit,
+	onDelete,
+} ) => (
+	<FormCheckbox.Label
+		as={ 'div' }
+		className={ className }
+		style={ useColorStyles( attributes ) }
+	>
+		<FormCheckbox type={ multipleChoice ? 'checkbox' : 'radio' } />
```
Comment (ice9js) [inline packages/block-editor/src/rating-scale-answer/edit-checkbox.js:27]: Hmm... I'm not sure why you'd want a multiple choice rating but feel free to enlighten me.

File: packages/block-editor/src/rating-scale-answer/style.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.wp-block[data-type="crowdsignal-forms/rating-scale-question"]:not(.is-selected):not(.has-child-selected) {
```
Comment (ice9js) [inline packages/block-editor/src/rating-scale-answer/style.scss:1]: Actually, this should go into the theme compatibility layer. If it's a default to be shared by all themes, put it in `base/editor.scss`.

File: packages/block-editor/src/rating-scale-answer/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ */
+import { MultipleChoiceAnswerIcon } from '@crowdsignal/icons';
+import attributes from './attributes';
+import EditRatingScaleAnswer from './edit';
+
+const name = 'crowdsignal-forms/rating-scale-answer';
+
+const settings = {
+	title: __( 'Answer', 'blocks' ),
+	description: __( 'Add more answer options to your question', 'blocks' ),
```
Comment (ice9js) [inline packages/block-editor/src/rating-scale-answer/index.js:17]: Nit: `i18n` domain doesn't match the package name.

File: packages/block-editor/src/rating-scale-question/attributes.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+		default: 5,
+	},
+	// Style attributes, should follow the name scheme supported by @crowdsignal/styles helpers.
+	backgroundColor: {
+		type: 'string',
+	},
+	borderColor: {
+		type: 'string',
+	},
+	borderRadius: {
```
Comment (ice9js) [inline packages/block-editor/src/rating-scale-question/attributes.js:26]: I believe the border radius and width props should default to `''` since #172.

File: packages/block-editor/src/rating-scale-question/edit.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+ * Internal dependencies
+ */
+import { useClientId } from '@crowdsignal/hooks';
+
+/**
+ * Style dependencies
+ */
+import { QuestionHeader, QuestionWrapper } from '@crowdsignal/blocks';
+// import Toolbar from './toolbar';
+import Sidebar from './sidebar';
```
Comment (ice9js) [inline packages/block-editor/src/rating-scale-question/edit.js:18]: I don't think these are style dependencies?

File: packages/blocks/src/components/question-wrapper/index.js [hunk 8]
```diff
@@ -56,6 +56,12 @@ const Content = styled.div`
 			margin-bottom: 0;
 		}
 	}
+
+	&.is-horizontal {
```
Comment (ice9js) [inline packages/blocks/src/components/question-wrapper/index.js:60]: Technically this works... I wonder though if it's maybe better to create a separate `<Row />` component for this or something? Would make adjusting some of these like `justify-content` easier in the future.  
And this will definitely not cut it for something like a matrix setup.

File: packages/blocks/src/rating-scale-answer/button.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+ */
+import { RawHTML } from '@wordpress/element';
+import styled from '@emotion/styled';
+
+/**
+ * Internal dependencies
+ */
+import { Button, FormCheckbox } from '../components';
+
+const ButtonContent = styled.span`
```
Comment (ice9js) [inline packages/blocks/src/rating-scale-answer/button.js:12]: As this is basically copied over from `multiple-choice-answer`, we should think about making this into its own thing maybe.
But could be considered outside of the scope for this PR.

File: packages/theme-compatibility/src/base/editor.scss [hunk 10]
```diff
@@ -53,6 +53,17 @@
 	}
 
 	.crowdsignal-forms-question-wrapper__content {
+
+		&.is-horizontal {
+			display: block; // abort flex to get full width!
```
Comment (ice9js) [inline packages/theme-compatibility/src/base/editor.scss:58]: Or `flex: 1;` ? Also, why is this necessary here? Shouldn't the styled component styles take care of it already?

**Review by ice9js — CHANGES_REQUESTED** [review]
Overall, I think it looks nice. I left a few small comments and I will need an explanation from someone about how a multiple choice rating makes sense?

File: packages/block-editor/src/rating-scale-answer/style.scss (hunk 4 above)
Comment (jcheringer) [reply packages/block-editor/src/rating-scale-answer/style.scss:1]: I'd ask why this is necessary?

**Comment by CGastrell** [comment]
Except for the style and row comments, all the other *quirky* things (i18n domain, wrong import section, etc) come from the copy/paste from MC Question and Answer blocks. I'll try to address those on both instances.

As for the styles: I think some `Row` thing makes sense. Not sure if that would only be for a matrix or it's already justified here. It's also true that when I faced the idea of this to *pave the road* for the matrix, a lot of things would need to be changed, specially some row/table behaviors that I still need to figure out (and don't have anything to test with). Most challenging one being how to make *cells* of the `Row` to share the width their above/below  neighbors (this was also a thing on the legacy editor, but we ended up using a full table there).

File: packages/block-editor/src/rating-scale-answer/edit-checkbox.js (hunk 3 above)
Comment (CGastrell) [reply packages/block-editor/src/rating-scale-answer/edit-checkbox.js:27]: Matrixes do have a "multiple answer" option. While this is not the case for ratings, I'd thought we could keep that here if we are to use this component in the future.

Comment (CGastrell) [reply packages/block-editor/src/rating-scale-answer/edit-checkbox.js:27]: Then again, the file is not used by any component, so I might just remove the whole thing

**Review by jcheringer — CHANGES_REQUESTED** [review]
@CGastrell One little thing I notice is that when you add the Rating block on a blank project, it doesn't include the Submit button.
We need to map the new block on `apps/dashboard/src/components/editor/auto-submit-button.js`.
Other than that, looks good to me.
About the `is-horizontal` thing, if we really want to handle this now, I'd do it on a separate PR.

**Comment by digitalwaveride** [comment]
Nice nice nice Christian!!!

The only thing ....  that emoji styling would be a very nice touch to have... 

image

**Review by jcheringer — APPROVED** [review]
LGTM!
